07.ls_object/terminal_display.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class TerminalDisplay
+  MAX_COLUMN = 3
+
+  def initialize(options)
+    @options = options
+    paths = @options.include?(:a) ? Dir.entries('.').sort : Dir.glob('*')
+    contents = paths.map { |path| Content.new(path) }
+    @contents = @options.include?(:r) ? contents.reverse : contents
+  end
+
+  def output
+    if @options.include?(:l)
+      output_detail_lines
+    else
+      output_pathname
+    end
+  end
+
+  private
+
+  def output_detail_lines
+    total_block_size = @contents.sum(&:blocks)
+    max_hardlink = @contents.map { |content| content.hardlink.to_s.size }.max
+    owner_name_width = @contents.map { |content| content.owner_name.length }.max
+    group_name_width = @contents.map { |content| content.group_name.length }.max
+    max_bytesize = @contents.map { |content| content.bytesize.to_s.size }.max
+
+    puts "total #{total_block_size}"
+    @contents.each do |content|
```

Comment:
元のコードのような書き方も悪くないですが、変数の寿命を極力短くするならこんな書き方もあります。

```suggestion
    total_block_size = @contents.sum(&:blocks)
    puts "total #{total_block_size}"
    
    max_hardlink = @contents.map { |content| content.hardlink.to_s.size }.max
    owner_name_width = @contents.map { |content| content.owner_name.length }.max
    group_name_width = @contents.map { |content| content.group_name.length }.max
    max_bytesize = @contents.map { |content| content.bytesize.to_s.size }.max
    @contents.each do |content|
```


あと変数名がmaxとwidthの2種類になってますが、全部widthで統一していいんじゃないでしょうか。
たとえば`max_bytesize`という名前だと、"10200545734bytes"みたいな最大のバイトサイズが入ってると勘違いされるかもしれません。